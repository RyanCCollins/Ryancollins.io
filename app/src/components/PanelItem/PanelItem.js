import React, { PropTypes } from 'react';
import { Column } from 'react-foundation';
import './PanelItem.scss';

const PanelItem = ({
  item,
  icon
}) => (
  <Column
    small={12}
    medium={6}
    large={4}
    className={
      `panel-item ${item.id > 2 ?
        'bottom' : 'top'} ${item.id !== 2 && item.id !== 5 ?
        'bordered' : 'not-bordered'}`
    }
  >
    <div className="inner-content">
      {icon}
      <h4
        className="centered raleway uppercase ribbon-text"
      >
        <span>{item.title}</span>
      </h4>
      <p className="text-small text-light justified">{item.description}</p>
    </div>
  </Column>
);

PanelItem.propTypes = {
  icon: React.Children,
  item: PropTypes.object.isRequired
};

export default PanelItem;
