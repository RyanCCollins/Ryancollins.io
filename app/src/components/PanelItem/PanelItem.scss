.panel-item {
  position: relative;
  padding: 15px 28px 25px;
}

.panel-item > .inner-content {
  display: flex;
  flex-direction: column;
  background-color: #fff;
  padding: 1em;
  width: 100%;
}

.panel-item > .inner-content p {
  flex: 1 0 auto;
}

.panel-item.bordered:after {
  @media screen and (min-width: 1025px) {
    position: absolute;
    content: '';
    top: 0;
    right: 0;
    width: 1px;
    height: 100%;
    border-right: 1px solid rgb(153, 13, 152);
  }
}

.panel-item:before {
  @media screen and (max-width: 1025px) {
    position: absolute;
    content: '';
    top: 0;
    right: 0;
    width: 1px;
    height: 100%;
    border-right: 1px solid rgb(153, 13, 152);
  }
}


.panel-item:after {
  @media screen and (max-width: 638px) {
    position: absolute;
    content: '';
    top: 0;
    right: 0;
    width: 1px;
    height: 100%;
    border-bottom: 1px solid rgb(153, 13, 152);
  }
}

.panel-item.top:before {
  @media screen and (min-width: 1025px) {
    position: absolute;
    content: '';
    bottom: 0;
    left: 0;
    width: 100%;
    height: 1px;
    border-bottom: 1px dashed rgb(153, 13, 152);
  }
}

.ribbon-text {
  position: relative;
  display: inline-block;
  margin: 0 40px 16px;
  color: white;
}

.ribbon-text:before {
  content: '';
  position: absolute;
  top: -9px;
  left: -44px;
  right: -44px;
  height: 44px;
  background: rgb(153, 13, 152);
  transform: rotate(-2deg);
  cursor:grab;
}

.ribbon-text:hover:before {
  transform: rotate(2deg);
}

.ribbon-text span {
  display: inline-block;
  position: relative;
}
