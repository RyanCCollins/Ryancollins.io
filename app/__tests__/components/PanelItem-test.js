import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import PanelItem from '../../src/components/PanelItem/PanelItem';
import FakeIcon from '../__mocks__/iconMock';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('<PanelItem />', () => {
  it('renders with default props', () => {
    const mockItem = {
      id: 1,
      title: 'Hello World',
      description: 'Lorem ipsum'
    };
    const component = renderer.create(
      <PanelItem item={mockItem} icon={FakeIcon()} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
