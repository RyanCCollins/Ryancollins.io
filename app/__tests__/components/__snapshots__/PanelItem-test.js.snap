exports[`<PanelItem /> renders with default props 1`] = `
<div
  className="columns panel-item
        top
        bordered small-12 medium-6 large-4"
  large={4}
  medium={6}
  small={12}>
  <div
    className="inner-content">
    <svg
      height="24px"
      role="img"
      version="1.1"
      viewBox="0 0 24 24.3616"
      width="24px" />
    <h4
      className="centered raleway uppercase ribbon-text">
      <span>
        Hello World
      </span>
    </h4>
    <p
      className="text-small text-light justified">
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Lorem ipsum"
          }
        } />
    </p>
  </div>
</div>
`;
